R/eval-walk.R

```diff
@@ -125,9 +129,21 @@ ensure_named <- function(pos,
   pos
 }
 
-check_empty <- function(x, allow_empty = TRUE, call = caller_env()) {
+check_empty <- function(x, allow_empty = TRUE, error_arg = NULL, call = caller_env()) {
   if (!allow_empty && length(x) == 0) {
-    cli::cli_abort("Must select at least one item.", call = call)
+    if (is.null(error_arg)) {
+      cli::cli_abort(
+        "Must select at least one item.",
+        call = call,
+        class = "tidyselect:::error_disallowed_empty"
+      )
+    } else {
+      cli::cli_abort(
+        "{.or {.arg {error_arg}}} must select at least one column.",
```

Comment:
I _think_ I'd use `.and` here? Not to imply that each argument must produce a selection, but to mean that they collectively must produce a selection? No strong opinion.